### PR TITLE
Add memcheck and check target

### PR DESCRIPTION
**Pull request by perrettecl**
* Rename coverage file to test-tools
	* Add memcheck global target (valgrind + tests)
	* Add check global target (tests)

### CODE REVIEW

File: cmake/test-tools.cmake [hunk 1]
```diff
@@ -14,88 +17,23 @@ if (NOT TARGET coverage)
...
+        COMMAND sh "${CMAKE_BINARY_DIR}/memcheck.targets"
+        COMMENT "Run valgrind on tests"
     )
+    file(WRITE ${CMAKE_BINARY_DIR}/memcheck.targets "")
+endif()
 
-    target_link_libraries(${newTarget} PRIVATE
-        ${linkLibs}
+if (NOT TARGET check)
+    add_custom_target(check
```
Comment (etnKlendathu) [inline cmake/test-tools.cmake:29]: I see only one suspicion, when you want to check few targets in one subtree... here could be collision with names.

Comment (perrettecl) [reply cmake/test-tools.cmake:29]: Should be the same with coverage, coverage-local then ?

Comment (etnKlendathu) [reply cmake/test-tools.cmake:29]: Yes, you might right. Let's try it.

**Review by etnKlendathu — APPROVED** [review]

**Review by FrancoisRegisDegott-eaton — APPROVED** [review]

File: cmake/fty-cmake.cmake [hunk 2]
```diff
@@ -18,11 +18,9 @@ include(${CMAKE_CURRENT_LIST_DIR}/external.cmake)
 # Warnings
 include(${CMAKE_CURRENT_LIST_DIR}/warnings.cmake)
 
-# Coverage
-include(${CMAKE_CURRENT_LIST_DIR}/coverage.cmake)
+# Tests tools include coverage and meme check
```
Comment (aquette) [inline cmake/fty-cmake.cmake]: typo on mem*e* ;)

**Review by aquette — APPROVED** [review]

**Pull request merged**